source/Octopus.Client/Model/RetentionDefaultConfigurationResource.cs

```diff
@@ -0,0 +1,10 @@
+﻿using Octopus.Client.Extensibility.Attributes;
+
+namespace Octopus.Client.Model
+{
+    public class RetentionDefaultConfigurationResource : Resource
```

Comment:
This class is going to need to set the `Id` property to a well known value. Similar to https://github.com/OctopusDeploy/Octofront/blob/f59e858ac93ed7e003e2b253029b3baaed961559/source/Octofront.Core/Hosted/Instances/DynamicWorkerConfiguration.cs#L5

Reply:
Why do we need to specify the id here?  We have a dedicated API endpoint for this resource: https://github.com/OctopusDeploy/OctopusDeploy/blob/le/file-retention/source/Octopus.Server/Web/Controllers/Configuration/Retention/ModifyRetentionDefaultConfigurationController.cs#L39.

I think this is not the case for DW which is an exception here.

Reply:
To call this endpoint would require a new repository in Octopus.Client. By specifying the Id on the Resource model, you can go via the existing Configuration Repository and don't actually need the new controller in Server. 
Example: https://octopus.com/docs/administration/managing-infrastructure/server-configuration/managing-configuration#net-c
And the corresponding Client Resource https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Client/Model/WebPortalConfigurationResource.cs

Reply:
We need a controller because we have some business rules to enforce. It's not a simple CRUD thing.

Reply:
Similar to how License is treated. It lives in the Configuration table but has its own endpoints.

Reply:
Just to clarify this one as this was discussed in this mornings sync, we are talking about 2 different repositories here.

The one that requires the Id to be set is the general purpose `Configuration` repository while for this particular case we can just use the `Client` repository instead. The client repository allows us to go to a specific/custom endpoint while `Configuration` has its own endpoint.

Example of this already being done from the [MaintenanceConfigurationResource](https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Client/Model/MaintenanceConfigurationResource.cs):

https://github.com/OctopusDeploy/Octofront/blob/master/source/Octofront.Core/Hosted/Octopus/MaintenanceModeConfigurator.cs#L45